sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/CBDE/PreservingEncodingTest.cs

```diff
@@ -0,0 +1,45 @@
+﻿extern alias csharp;
+
+using System;
+using System.Text;
+using csharp::SonarAnalyzer.CBDE;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace SonarAnalyzer.UnitTest.CBDE
+{
+    [TestClass]
+    public class PreservingEncodingTest
```

Comment:
could you also add a test with a `.` inside the name? 

if it throws, you can use `[ExpectedException(typeof(NotSupportedException))]` like in https://github.com/SonarSource/sonar-dotnet/blob/b68f6175b8504a670bc6e3a9e1e650c08fac69fe/sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Common/AnalyzerLanguageTest.cs#L39

Reply:
I had it first, then removed it because I did not feel it was valuable... Adding it back.

Reply:
I believe it's good to document via tests unhappy-path behavior 😄